gerritlab/merge_request.py

```diff
@@ -83,7 +83,7 @@ def create(self):
 
     def update(
             self, source_branch=None, target_branch=None, title=None,
-            description=None):
+            description=None, state_event=None):
```

Comment:
Nitpick: there are already a lot of arguments to this function—a smell that might indicate need for a refactor.

As a first (maybe terrible) thought, MergeRequest.update could accept a MergeRequest object as an argument(?)

Reply:
For clarity: not needed for this to land. An idea for a future PR.

Reply:
Yes this class definitely needs a refactor. There are a lot other merge request attributes that we currently don’t support. I was thinking maybe we can keep track of all the attributes in a dict and override __getattr__()/__setattr()__ to provide easy getters/setters. So we can change attributes such as changing the MR title by setting the title attribute and the update() method doesn’t need any argument.

Reply:
I was thinking about overriding `__setattr__()` to provide setters for any merge request attribute so we don't need to add a getter/setter for each one.

Reply:
I like that idea.